x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -376,34 +379,38 @@ protected void onStart(long now, ActionListener<Boolean> listener) {
     protected void initializeFunction() {
         // create the function
         function = FunctionFactory.create(getConfig());
+        List<String> warnings = getWarnings(function, getConfig().getSource(), getConfig().getSyncConfig());
+        for (String warning : warnings) {
+            logger.warn(new ParameterizedMessage("[{}] {}", getJobId(), warning));
+            auditor.warning(getJobId(), warning);
+        }
```

Comment:
This will happen at every checkpoint. This will end up spamming the auditor log. 

I think this should probably only ever be written when the transform initially starts and maybe when it is assigned to a node.

Reply:
I've moved the `logger.warn` and `auditor.warning` calls from `initializeFunction` to the `if (initialRun())` branch.
That should solve the issue.

Reply:
Unfortunately, my manual tests showed the warnings are still emitted for every checkpoint so please hold on with the review.

Reply:
Ok, I moved the `auditor.warning` calls to actions (Put, Update) so that they are not executed on every checkpoint.
PTAL @benwtrent and @hendrikmuhs